ArchiSteamFarm/Commands.cs

```diff
@@ -1813,6 +1825,160 @@ public sealed class Commands {
+			if ((steamID == 0) || !new SteamID(steamID).IsIndividualAccount) {
+				throw new ArgumentOutOfRangeException(nameof(steamID));
+			}
+
+			if (string.IsNullOrEmpty(botNames)) {
+				throw new ArgumentNullException(nameof(botNames));
+			}
+
+			HashSet<Bot>? bots = Bot.GetBots(botNames);
+
+			if ((bots == null) || (bots.Count == 0)) {
+				return ASF.IsOwner(steamID) ? FormatStaticResponse(string.Format(CultureInfo.CurrentCulture, Strings.BotNotFound, botNames)) : null;
+			}
+
+			IList<string?> results = await Utilities.InParallel(bots.Select(bot => Task.Run(() => bot.Commands.ResponseMatchActivelyBlacklist(steamID)))).ConfigureAwait(false);
+
+			List<string> responses = new(results.Where(result => !string.IsNullOrEmpty(result))!);
+
+			return responses.Count > 0 ? string.Join(Environment.NewLine, responses) : null;
+		}
+
+		private string? ResponseMatchActivelyBlacklistAdd(ulong steamID, string targetAppIDs) {
+			if ((steamID == 0) || !new SteamID(steamID).IsIndividualAccount) {
+				throw new ArgumentOutOfRangeException(nameof(steamID));
+			}
+
+			if (string.IsNullOrEmpty(targetAppIDs)) {
+				throw new ArgumentNullException(nameof(targetAppIDs));
+			}
+
+			if (!Bot.HasAccess(steamID, BotConfig.EAccess.Master)) {
+				return null;
+			}
+
+			string[] targets = targetAppIDs.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+
+			if (targets.Length == 0) {
+				return FormatBotResponse(string.Format(CultureInfo.CurrentCulture, Strings.ErrorIsEmpty, nameof(targets)));
+			}
+
+			HashSet<uint> appIDs = new();
+
+			foreach (string target in targets) {
+				if (!uint.TryParse(target, out uint appID) || (appID == 0)) {
+					return FormatBotResponse(string.Format(CultureInfo.CurrentCulture, Strings.ErrorParsingObject, nameof(appID)));
+				}
+
+				appIDs.Add(appID);
+			}
+
+			if (!Bot.BotDatabase.IdlingBlacklistedAppIDs.AddRange(appIDs)) {
+				return FormatBotResponse(Strings.NothingFound);
+			}
+
+			return FormatBotResponse(Strings.Done);
+		}
+
+		private static async Task<string?> ResponseMatchActivelyBlacklistAdd(ulong steamID, string botNames, string targetAppIDs) {
+			if ((steamID == 0) || !new SteamID(steamID).IsIndividualAccount) {
+				throw new ArgumentOutOfRangeException(nameof(steamID));
+			}
+
+			if (string.IsNullOrEmpty(botNames)) {
+				throw new ArgumentNullException(nameof(botNames));
+			}
+
+			if (string.IsNullOrEmpty(targetAppIDs)) {
+				throw new ArgumentNullException(nameof(targetAppIDs));
+			}
+
+			HashSet<Bot>? bots = Bot.GetBots(botNames);
+
+			if ((bots == null) || (bots.Count == 0)) {
+				return ASF.IsOwner(steamID) ? FormatStaticResponse(string.Format(CultureInfo.CurrentCulture, Strings.BotNotFound, botNames)) : null;
+			}
+
+			IList<string?> results = await Utilities.InParallel(bots.Select(bot => Task.Run(() => bot.Commands.ResponseMatchActivelyBlacklistAdd(steamID, targetAppIDs)))).ConfigureAwait(false);
```

Comment:
This way `targetAppIDs` is split and parsed into ulongs once for each bot. We could create one method `ResponseMatchActivelyBlacklistAdd(ulong steamID, HashSet<uint> appIDs targetAppIDs)` and extract this logic.

Reply:
It follows the same logic as in `ResponseIdleBlacklistAdd` and `ResponseBlacklistAdd` - if this should be really changed (parsing is not really expensive operation), probably it should be changed in all the places.